### DISCUSSION TITLE
How do I display another TUI program within a block? ie vim, cfm, etc.

### DISCUSSION THREAD

**Discussion by GVodyanov** [fdehau/tui-rs#549] [Q&A]
I can't find anything in the docs and no one on reddit seems to know either, [link to post](https://www.reddit.com/r/learnrust/comments/qca2vm/how_to_display_another_tui_program_in_a_block_in/)

**Comment by PierreAntoineGuillaume** [comment]
I guess I'd start with executing it in a spawned command, and piping its IO.

On paper it seems pretty easy, but i wouldn't trust that feeling, because to display a tui you need to enter raw mode, and the underlaying tui wil maybe want to do the same and I don't know if the raw mode behaves like a stack or a global state carried by the terminal, and I expect it to be carried by the terminal ; given how a panic when raw mode is active leaves the terminal in a broken state.

And maybe you'd have to wrap the input, and the mouse as well ; and maybe provide the wrapped ui a fake mouse.

https://docs.rs/crossterm/0.3.0/crossterm/raw/index.html
